app/flatpak-builtins-build.c

```diff
@@ -319,10 +298,9 @@ flatpak_builtin_build (int argc, char **argv, GCancellable *cancellable, GError
       extension_point = g_build_filename (bare_extension_point, x_subdir_suffix, NULL);
     }
 
-  argv_array = g_ptr_array_new_with_free_func (g_free);
-  fd_array = g_array_new (FALSE, TRUE, sizeof (int));
-  g_array_set_clear_func (fd_array, clear_fd);
-  g_ptr_array_add (argv_array, g_strdup (flatpak_get_bwrap ()));
+  minimal_envp = flatpak_run_get_minimal_env (TRUE, FALSE);
```

Comment:
Not related to the PR, but a function interface with two anonymous booleans is a bit iffy, no offense.  Actually I think you (so far) do not call it with ld_so_cache = TRUE?  Do you have a plan for that or would you take a patch to drop some of that code?

Reply:
It is called with TRUE in flatpak_run_app(), but yeah, some kind of flags instead would be a nice cleanup. In fact, all the run code is pretty fugly, and could use a cleanup (of which this patch is a part.)

Any further cleanup help would be appreciated, but we need to be very careful so we don't change things breaking the sandbox or changing its behaviour.

Reply:
> It is called with TRUE in flatpak_run_app(),

Maybe you mean the call to **flatpak_run_apply_env_default**() from **flatpak_run_app**()?

I didn't see **flatpak_run_get_minimal_env**() called with use_ld_so_cache = TRUE. (Not yet)

I'm not following all the logic but since they are similar in your mind, there is probably potential to reduce duplication even more and then this might even sort itself out.

In any case I get it, the cleanup is work-in-progress.